pkg/imgpkg/registry/keychain.go

```diff
@@ -20,7 +22,11 @@ type KeychainOpts struct {
 }
 
 func Keychain(keychainOpts KeychainOpts, environFunc func() []string) regauthn.Keychain {
-	return regauthn.NewMultiKeychain(&envKeychain{environFunc: environFunc}, customRegistryKeychain{opts: keychainOpts})
+	k8sKeychain, err := k8schain.NewFromPullSecrets(context.Background(), nil)
+	if err != nil {
+		panic(err.Error())
+	}
+	return regauthn.NewMultiKeychain(k8sKeychain, &envKeychain{environFunc: environFunc}, customRegistryKeychain{opts: keychainOpts})
```

Comment:
something to think about:
- where should k8s keychain be inserted. should credentials provided explicitly take precedence over implicit creds?
- it seems that k8s keychain does quite a bit extra -- it does k8s specific lookups in addition to iaas provided creds. i dont think we want to include k8s lookups.
- ive been clicking around and stumbled upon https://github.com/kubernetes/kubernetes/issues/86245 along side https://github.com/vdemeester/k8s-pkg-credentialprovider/issues.

Reply:
> should credentials provided explicitly take precedence over implicit creds

The order is going to be env -> iaas -> cli -> config.json

iaas takes precedence over cli because iaas is targeted to a registry hostname (while cli isn't)

> k8s keychain does quite a bit extra

True, I removed the k8s keychain and instead will register the iaas providers via *magic imports*. Also this reduces the number of vendor files. And makes testing a bit simpler too.

> ive been clicking around and stumbled upon ...

good catch. I have added logic to avoid waiting forever during initializing the iaas keychain :-) 

Also, I added retry logic when getting the `Authorization` credentials from the metadata service too.